src/com/serotonin/mango/rt/dataImage/PointValueCacheCooperateWithPointValueDao.java

```diff
@@ -0,0 +1,68 @@
+package com.serotonin.mango.rt.dataImage;
+
+import com.serotonin.mango.db.dao.PointValueDao;
+import com.serotonin.mango.rt.dataImage.exceptions_level.ExceptionsArea;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+class PointValueCacheCooperateWithPointValueDao {
+
+    private PointValueDao pointValueDao;
+
+    PointValueTime savePointValueIntoDatabaseAsyncOrSync(
+            PointValueTime pointValueTime,
+            SetPointSource pointSource,
+            boolean async,
+            int dataPointId){
+
+        if (async)
+            getPointValueDao().savePointValueAsync(dataPointId, pointValueTime, pointSource);
+        else
+            pointValueTime = getPointValueDao().savePointValueSync(dataPointId, pointValueTime, pointSource);
+
+        return pointValueTime;
+    }
+
+    void logPointValueAsync(PointValueTime pointValue, SetPointSource source, int dataPointId) {
+        // Save the new value and get a point value time back that has the id and annotations set, as appropriate.
+        getPointValueDao().savePointValueAsync(dataPointId, pointValue, source);
+
+    }
+
+    PointValueTime getLatestPointValueFromDao(int dataPointId) {
+
+        return getPointValueDao().getLatestPointValue( dataPointId );
+
+    }
+
+    LinkedList<PointValueTime> getDefinedLimitRowsOfLatestPointValues(int dataPointId, int limit) {
+
+        LinkedList<PointValueTime> pointValueTimes = new LinkedList<PointValueTime>();
+
+        for(PointValueTime pointValueTime: getLatestPointValuesByDataPointIdAndLimit(dataPointId,limit)){
+            pointValueTimes.addFirst(pointValueTime);
+        }
+
+        return pointValueTimes;
+    }
+
+    private List<PointValueTime> getLatestPointValuesByDataPointIdAndLimit(int dataPointId,int limit) {
+
+        List<PointValueTime> pointValueTimes = getPointValueDao().getLatestPointValues(dataPointId,limit);
+
+        return (new ExceptionsArea().doListIsNull( pointValueTimes ))
+                ?Collections.emptyList()
+                :pointValueTimes;
```

Comment:
In my opinion, it can't be a null, but even if it would be better to add a logger and do something like this:

```
        List<PointValueTime> pointValueTimes = getPointValueDao().getLatestPointValues(dataPointId,limit);

        if(pointValueTimes == null) {
            log.warn("");
            return Collections.emptyList();
        }

        return pointValueTimes;
```
Class ExceptionsArea, in my opinion, is deleted, this method does two things, checks if the class is null and returns an exception that is immediately caught and returns boolean. It is not clear what you wanted to achieve.

Reply:
That class has been created to separate exception area from another logic in application. I wrote some times ago about layers ( logic, exceptions, dao,DAO,fasade and different).

Reply:
But see the exception is not necessary here ... :)

Reply:
But look at my example here there is no need to return an exception. :)